src/Compilers/CSharp/Portable/Symbols/Attributes/RetargetingAttributeData.cs

```diff
@@ -4,43 +4,52 @@
+        private readonly ImmutableArray<KeyValuePair<string, TypedConstant>> _namedArguments;
+
         internal RetargetingAttributeData(
-            SyntaxReference applicationNode,
+            CSharpAttributeData underlying,
             NamedTypeSymbol attributeClass,
-            MethodSymbol attributeConstructor,
+            MethodSymbol? attributeConstructor,
             ImmutableArray<TypedConstant> constructorArguments,
-            ImmutableArray<int> constructorArgumentsSourceIndices,
-            ImmutableArray<KeyValuePair<string, TypedConstant>> namedArguments,
-            bool hasErrors,
-            bool isConditionallyOmitted)
-            : base(applicationNode, attributeClass, attributeConstructor, constructorArguments, constructorArgumentsSourceIndices, namedArguments, hasErrors, isConditionallyOmitted)
+            ImmutableArray<KeyValuePair<string, TypedConstant>> namedArguments)
         {
+            Debug.Assert(underlying is SourceAttributeData or SynthesizedAttributeData);
+
+            _underlying = underlying;
+            _attributeClass = attributeClass;
+            _attributeConstructor = attributeConstructor;
+            _constructorArguments = constructorArguments;
+            _namedArguments = namedArguments;
         }
 
-        /// <summary>
-        /// Gets the retargeted System.Type type symbol.
-        /// </summary>
-        /// <param name="targetSymbol">Target symbol on which this attribute is applied.</param>
-        /// <returns>Retargeted System.Type type symbol.</returns>
-        internal override TypeSymbol GetSystemType(Symbol targetSymbol)
-        {
-            var retargetingAssembly = (RetargetingAssemblySymbol)(targetSymbol.Kind == SymbolKind.Assembly ? targetSymbol : targetSymbol.ContainingAssembly);
-            var underlyingAssembly = (SourceAssemblySymbol)retargetingAssembly.UnderlyingAssembly;
+        public override NamedTypeSymbol AttributeClass => _attributeClass;
+        public override MethodSymbol? AttributeConstructor => _attributeConstructor;
+        protected internal override ImmutableArray<TypedConstant> CommonConstructorArguments => _constructorArguments;
+        protected internal override ImmutableArray<KeyValuePair<string, TypedConstant>> CommonNamedArguments => _namedArguments;
 
-            // Get the System.Type from the underlying assembly's Compilation
-            TypeSymbol systemType = underlyingAssembly.DeclaringCompilation.GetWellKnownType(WellKnownType.System_Type);
+        public override SyntaxReference? ApplicationSyntaxReference => null;
 
-            // Retarget the type
-            var retargetingModule = (RetargetingModuleSymbol)retargetingAssembly.Modules[0];
-            return retargetingModule.RetargetingTranslator.Retarget(systemType, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
-        }
+        [MemberNotNullWhen(true, nameof(AttributeClass), nameof(AttributeConstructor))]
```

Comment:
If `_attributeConstructor` is `null`, then `HasErrors` is true. So it looks like the nullability attribute/hint is wrong.
Also, `AttributeClass` is always not-null, so we shouldn't need to specify it in the attribute.

Reply:
This is a pre-existing condition from the base class, please open a separate issue.

Reply:
Opened #70555